### ISSUE TITLE
Questions about Relative Range AbsRel and Relative Range a1 metric

### ISSUE THREAD

**Issue by waterbearbee** [SeasonDepth/SeasonDepth#12]
Thanks for your work!
We have doubts about Relative Range AbsRel and Relative Range a1 metric. We believe that these two metric can not well reflect the performance of the model. For example, we have two sets of experimental results, one is R1, and the other one is R2. We use the official evaluation code to evaluate on the validation set, and the results are as follows. 

R1 (Mean a1 for each env): 
{env00: {a1: 0.30}, env01: {a1: 0.31}, env02: {a1: 0.30}, env03: {a1: 0.34}, env04: {a1: 0.31}, env05: {a1: 0.33}, env06: {a1: 0.32}, env07: {a1: 0.37}, env08: {a1: 0.31}, env09: {a1: 0.27}, env10: {a1: 0.30}, env11: {a1: 0.30}}
R1 (Mean a1 for all envs): 0.32
R1 (Relative Range a1 for all envs): 0.14

R2 (Mean a1 for each env): 
{env00: {a1: 0.66}, env01: {a1: 0.70}, env02: {a1: 0.73}, env03: {a1: 0.72}, env04: {a1: 0.71}, env05: {a1: 0.76}, env06: {a1: 0.68}, env07: {a1: 0.71}, env08: {a1: 0.72}, env09: {a1: 0.68}, env10: {a1: 0.70}, env11: {a1: 0.71}}
R2 (Mean a1 for all envs): 0.70
R2 (Relative Range a1 for all envs): 0.22

Obviously, the Mean a1 metric of R2 is much better than that of R1 for each env. Therefore, generally speaking, the R2 model must be better than the R1 model. However, we can find that the Relative Range a1 metric of R2 is worse than that of R1, which is unreasonable.

Therefore, we believe that Relative Range AbsRel and Relative Range a1 metric can not reflect the performance of the model.
 And these two metrics should be eliminated from the competition metric.
 
We want to know, how do you think about this problem?

**Comment by yangbaoquan** [comment]
Thanks for your question.

In fact, the metrics rng_absrel and rng_a1 are used to evaluate the robustness of prediction output. It should be noted that we mainly focus on which models perform better on all environments we've provided, those extreme envs in particular. Therefore, these two metrics cannot be removed from final score in this competition.

As you mentioned in your comment, it is true that the higher avg_a1 is, the higher rng_a1, rng_{1-a1} actually, tends to be. For example, rng of {0,1,2} is obviously higher than that of {60,61,62}. 

In conclusion, we will consider adjusting the weight of final_score. You can check the new ranking method in this repo soon.

Best.

**Comment by waterbearbee** [comment]
In fact, Variance AbsRel and Variance a1 metrics can also reflect the robustness of the model, and the above problems can be avoided to a certain extent.

**Comment by yangbaoquan** [comment]
> In fact, Variance AbsRel and Variance a1 metrics can also reflect the robustness of the model.

Yes, you are right. While as we mentioned in our paper, **considering the depth prediction applications, it should be more rigorous to prevent better results fluctuation than worse results under changing conditions**. *Relative range* is a more strict matric than *Variance*, because it's more sensitive to reflect robustness under changing environment, which is important for safe AI.

We admit that as for a result performing well in *Avg* and another performing badly, *Rng* of the former tends to be greater. Therefore, we will consider lower the weight of *Rng* instead of removing it.

Hope my reply above can dismiss your concern.

**Comment by waterbearbee** [comment]
Thanks for your reply,  I see.
